File: app/src/main/java/ua/com/amicablesoft/android/wr/dal/IRepository.java
package ua.com.amicablesoft.android.wr.dal;

import java.util.ArrayList;

import ua.com.amicablesoft.android.wr.models.Competition;
import ua.com.amicablesoft.android.wr.models.Powerlifter;
import ua.com.amicablesoft.android.wr.models.User;

/**
 * Created by lapa on 07.10.16.
 */
public interface IRepository {

    interface LoadUserCallback {
        void found(boolean userFound);
    }

    interface LoadPowerliftersCallback {
        void onPowerliftersLoaded(ArrayList<Powerlifter> powerlifterArrayList);
        void onDataNotAvailable();
    }

    interface LoadCompetitionsCallback {
        void onCompetitionsLoaded(ArrayList<Competition> competitionArrayList);
        void onDataNotAvailable();
    }

    interface AddCompetitionCallback {
        void onCompetitionAddedSuccess();
    }

    void userExist(User user, LoadUserCallback loadUserCallback);
    void userSave(User user);

    void writeNewPowerlifter(String name, String lastName);
    void getPowerlifters(LoadPowerliftersCallback loadPowerliftersCallback);

    void writeNewCompetition(String competition, AddCompetitionCallback addCompetitionCallback);
    void getCompetitions(LoadCompetitionsCallback loadCompetitionsCallback);
}

File: app/src/main/java/ua/com/amicablesoft/android/wr/dal/IRepository.kt
/*
 * Created by Olha Tymoshenko on 5/24/18 8:34 PM.
 * Copyright (c) 2018. All rights reserved.
 */

package ua.com.amicablesoft.android.wr.dal

import ua.com.amicablesoft.android.wr.models.Competition
import ua.com.amicablesoft.android.wr.models.Powerlifter
import ua.com.amicablesoft.android.wr.models.User

interface IRepository {

    interface LoadUserCallback {
        fun found(userFound: Boolean?)
    }

    interface LoadPowerliftersCallback {
        fun onPowerliftersLoaded(powerlifterArrayList: List<Powerlifter?>)
        fun onDataNotAvailable()
    }

    interface LoadCompetitionsCallback {
        fun onCompetitionsLoaded(competitionArrayList: List<Competition?>)
        fun onDataNotAvailable()
    }

    fun userExist(user: User, loadUserCallback: LoadUserCallback)
    fun userSave(user: User)

    fun writeNewPowerlifter(name: String, lastName: String)
    fun getPowerlifters(loadPowerliftersCallback: LoadPowerliftersCallback)

    fun writeNewCompetition(competition: String)
    fun getCompetitions(loadCompetitionsCallback: LoadCompetitionsCallback)
}